### PR TITLE
[v7r0] Enable markdown files in docs

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
*Docs
NEW: add ``recommonmark`` extension to allow use of markdown files in the documenation

ENDRELEASENOTES

As an example of the markdown files, I changed the vmdirac documentation to point to vmdirac.readthedocs.io

https://diracsailer.readthedocs.io/en/fixdoc/AdministratorGuide/ServerInstallations/vmdirac.html

Some sphinx things are not possible to do in markdown, and anchors in references are tricky already.

### CODE REVIEW

File: docs/source/AdministratorGuide/ServerInstallations/vmdirac.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+```eval_rst
+.. _VMDIRAC:
+```
```
Comment (fstagni) [inline docs/source/AdministratorGuide/ServerInstallations/vmdirac.md]: What are the above for?

File: docs/source/conf.py [hunk 2]
```diff
@@ -76,13 +79,26 @@
...
+                       True)
+  app.add_transform(AutoStructify)
+
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
 
 # The suffix of source filenames.
-source_suffix = '.rst'
+source_suffix = {'.rst': 'restructuredtext',
+                 '.txt': 'ma',
```
Comment (fstagni) [inline docs/source/conf.py]: what is `'ma'` for?

File: docs/source/AdministratorGuide/ServerInstallations/vmdirac.md (hunk 1 above)
Comment (andresailer) [reply docs/source/AdministratorGuide/ServerInstallations/vmdirac.md]: This is to set an "anchor"

File: docs/source/conf.py (hunk 2 above)
Comment (andresailer) [reply docs/source/conf.py]: That is a typo,
```suggestion
                 '.txt': 'markdown',
```

**Comment by chaen** [comment]
I am a bit lost, why do we now mix markdown and rst ? what is the gain ?

**Comment by fstagni** [comment]
We don't HAVE to mix, and I would actually encourage everyone to use RST instead of MD, as RST is native to Sphinx (and a bit more powerful, although slightly more complicated). Anyway, there are cases of documentation ALREADY written (somewhere else) in MD that we want to simply add. This just gives the possibility for our doc to interpret it in case it arises. But all new docs should be written in RST.

**Comment by andresailer** [comment]
Do I have to add the new package `recommonmark` somewhere here ,or does this only work if it is part of diracos? I already added it to requirements.txt here

**Comment by chaen** [comment]
In that case, wouldn't it be better to just convert the MD to RST once for all? 

@andresailer the doc build does not use DIRACOS so I think it is fine where you have it

**Comment by andresailer** [comment]
but it fails:
```
docs/source/conf.py:23: [E0401(import-error), ] Unable to import 'recommonmark'
docs/source/conf.py:24: [E0401(import-error), ] Unable to import 'recommonmark.transform'
```

**Comment by chaen** [comment]
ah right, because github actions does not like `requirements.txt`.... add it to `environment.yml`

**Pull request merged**